zipline/pipeline/factors/technical.py

```diff
@@ -683,3 +680,117 @@ def compute(self, today, assets, out, highs, lows, closes):
+    signal_period' : int >= 0, <= slow_period
+        The window length for the signal line.
+
+    Returns
+    -------
+    MACD: The difference between "fast" EMA and "slow" EMA.
+    signal: The EMA of the MACD line using `signal_period` as span.
+    hist: Difference between MACD and signal. (Divergence series)
+    """
+    inputs = [USEquityPricing.close]
+    params = ('fast_period', 'slow_period', 'signal_period')
+    outputs = ('MACD', 'signal', 'hist')
+
+    def __new__(cls,
+                fast_period=12,
+                slow_period=26,
+                signal_period=9,
+                *args,
+                **kwargs):
+        return super(MovingAverageConvergenceDivergence, cls).__new__(
+            cls,
+            fast_period=fast_period,
+            slow_period=slow_period,
+            signal_period=signal_period,
+            window_length=slow_period + signal_period - 1,
+            *args, **kwargs
+        )
+
+    def calculate_ewa(self, data, length):
+        decay_rate = 1.0 - (2.0 / (1.0 + length))
+        return average(data,
+                       axis=1,
+                       weights=self.weights(length, decay_rate))
+
+    def rolling_window(self, a, window):
+        # used to calculate rolling statistics using np.array
+        shape = a.shape[:-1] + (a.shape[-1] - window + 1, window)
+        strides = a.strides + (a.strides[-1],)
+        return as_strided(a, shape=shape, strides=strides)
+
+    def calculate_macd(self, col):
+        try:
+            slow_EWA = self.calculate_ewa(
+                self.rolling_window(
+                    col,
+                    self.params['slow_period']
+                ),
+                self.params['slow_period'])
+            fast_EWA = self.calculate_ewa(
+                self.rolling_window(
+                    col,
+                    self.params['fast_period']
+                )[-self.params['signal_period']:],
+                self.params['fast_period'])
+            macd = fast_EWA - slow_EWA
+            signal_line = self.calculate_ewa(
+                macd.reshape(-1, self.params['signal_period']),
+                self.params['signal_period'])
+            hist = macd[-1] - signal_line
+            return macd[-1], signal_line[-1], hist[-1]
+        except:
+            return nan, nan, nan
+
+    def compute(self, today, assets, out, close, fast_period, slow_period,
+                signal_period):
+        macd, sig, hist = zip(*map(self.calculate_macd, close.T))
+        out.macd[:] = macd
+        out.signal[:] = sig
+        out.hist[:] = hist
+
+
+class AnnualVolatility(CustomFactor):
+    """
+    Volatility
+    https://en.wikipedia.org/wiki/Volatility_(finance)
+
+    The degree of variation of a series over time as measured by the standard
+    deviation of the data over the course of a year.
+
+    **Default Inputs:** :data:`zipline.pipeline.data.USEquityPricing.close`
+    """
+    inputs = [USEquityPricing.close]
```

Comment:
According to the wikipedia link cited: "In finance, volatility is the degree of variation of a trading price series over time as measured by the standard deviation of returns." Would this default value be better of as `Returns`?